pypeit/core/coadd.py

```diff
@@ -687,27 +687,42 @@ def interp_spec(wave_new, waves, fluxes, ivars, masks):
 
 
 def sn_weights(waves, fluxes, ivars, masks, sn_smooth_npix, const_weights=False,
-               ivar_weights=False, verbose=False):
+               ivar_weights=False, relative_weights=False, ref_spec=-1, verbose=False):
 
     """
     Calculate the S/N of each input spectrum and create an array of
     (S/N)^2 weights to be used for coadding.
 
     Args:
-        fluxes: float ndarray, shape = (nspec, nexp)
-            Stack of (nspec, nexp) spectra where nexp = number of
-            exposures, and nspec is the length of the spectrum.
-        ivars: float ndarray, shape = (nspec, nexp)
-            Inverse variance noise vectors for the spectra
-        masks: bool ndarray, shape = (nspec, nexp)
-            Mask for stack of spectra. True=Good, False=Bad.
-        waves: float ndarray, shape = (nspec,) or (nspec, nexp)
+        waves : float ndarray, shape = (nspec,) or (nspec, nexp)
             Reference wavelength grid for all the spectra. If wave is a
             1d array the routine will assume that all spectra are on the
             same wavelength grid. If wave is a 2-d array, it will use
             the individual
-        sn_smooth_npix: float, optional, default = 10000.0
+        fluxes : float ndarray, shape = (nspec, nexp)
+            Stack of (nspec, nexp) spectra where nexp = number of
+            exposures, and nspec is the length of the spectrum.
+        ivars : float ndarray, shape = (nspec, nexp)
+            Inverse variance noise vectors for the spectra
+        masks : bool ndarray, shape = (nspec, nexp)
+            Mask for stack of spectra. True=Good, False=Bad.
+        sn_smooth_npix : float
             Number of pixels used for determining smoothly varying S/N ratio weights.
+        const_weights : bool
+            Use a constant weights for each spectrum?
+        ivar_weights : bool
+            Use inverse variance weighted scheme?
+        relative_weights : bool
+            Perform relative weights? See ref_spec for further details
+        ref_spec : int
+            Index of spectrum (i.e. 0 <= ref_spec < nstack) to be used as a reference.
```

Comment:
Clarify that if ref_spec == -1, then the code automatically sets the reference spectrum to be the highest S/N ratio spectrum. Also make it clear this is only used if the relative weights option is True.